### PR TITLE
Add cidrMask on privnet

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
- [x] `privnet create`
- [x] `privnet update`

Fix https://github.com/exoscale/cli/issues/41

### CODE REVIEW

File: cmd/privnet_create.go [hunk 1]
```diff
@@ -133,6 +142,7 @@ func init() {
 	privnetCreateCmd.Flags().StringP("startip", "s", "", "the beginning IP address in the network IP range. Required for managed networks.")
 	privnetCreateCmd.Flags().StringP("endip", "e", "", "the ending IP address in the network IP range. Required for managed networks.")
 	privnetCreateCmd.Flags().StringP("netmask", "n", "", "the netmask of the network. Required for managed networks.")
+	privnetCreateCmd.Flags().IntP("cidrmask", "c", 0, "the cidrmask of the network. Required for managed networks.")
```
Comment (greut) [inline cmd/privnet_create.go]: 0 is a valid mask though

File: cmd/privnet_create.go [hunk 2]
```diff
@@ -37,6 +38,19 @@ var privnetCreateCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		cidrmask, err := cmd.Flags().GetString("cidrmask")
+		if err != nil {
+			return err
+		}
+		if netmask == "" && cidrmask != "" {
+			c, err := strconv.ParseInt(cidrmask, 10, 32)
```
[the remark this answers is not in the document]
Comment (greut) [reply cmd/privnet_create.go]: create/update are very much alike, we could probably factor out way more than that.

**Comment by greut** [comment]
(rebased)

File: cmd/privnet_create.go [hunk 3]
```diff
@@ -25,48 +23,61 @@ var privnetCreateCmd = &cobra.Command{
...
 		}
 
-		if zone == "" {
-			zone = gCurrentAccount.DefaultZone
+		if nmask != "" && cidrmask != "" {
+			return fmt.Errorf("netmask %q and cidrmask %q are mutually exclusive", nmask, cidrmask)
 		}
 
-		if zone == "" {
-			reader := bufio.NewReader(os.Stdin)
```
Comment (vincentbernat) [inline cmd/privnet_create.go]: This part has disappeared?

**Pull request merged**